### DISCUSSION TITLE
Does CodeQL support Incremental scan ?

### DISCUSSION THREAD

**Discussion by pranay-kumar-tirumala** [github/codeql#17886] [Q&A]
I am trying to find out whether CodeQL supports incremental builds so that we can save hours spent on redundant builds and analysis steps which produce already identified defects.

**Comment by mbg** [comment]
Hi @pranay-kumar-tirumala,

No, CodeQL does not currently support incremental scans. Depending on which language your project is in, there are various options for speeding up the builds though, such as e.g. `build-mode: none` for Java and C#.

**Reply by pranay-kumar-tirumala** [reply]
Mine is C++. Do we have something for that?

**Comment by intrigus-lgtm** [comment]
There is/was a GitHub Next project for incremental CodeQL, but it doesn't look like it ever went into production:
https://githubnext.com/projects/incremental-codeql/
